snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -860,32 +858,13 @@ def _handle_dependencies(self, all_dependencies: Set[str]):
         staged_dependency_paths = {os.path.dirname(d) for d in staged}
         dependency_paths = part_dependency_paths | staged_dependency_paths
 
-        if not self._build_attributes.no_system_libraries():
-            system_dependency_paths = {os.path.dirname(d) for d in system}
-            dependency_paths.update(system_dependency_paths)
-
-            if system:
-                # Lots of dependencies are linked with a symlink, so we need to
-                # make sure we follow those symlinks when we migrate the
-                # dependencies.
-                _migrate_files(
-                    system,
-                    system_dependency_paths,
-                    "/",
-                    self.primedir,
-                    follow_symlinks=True,
-                )
-                formatted_system = "\n".join(sorted(system))
-                logger.warning(
-                    "Files from the build host were migrated into the snap to "
-                    "satisfy dependencies that would otherwise not be met. "
-                    "This feature will be removed in a future release. If "
-                    "these libraries are needed in the final snap, ensure "
-                    "that the following are either satisfied by a "
-                    "stage-packages entry or through a part:\n{}".format(
-                        formatted_system
-                    )
-                )
+        if system:
+            formatted_system = "\n".join(sorted(system))
+            # We cannot error if we consider the content interface...
+            logger.warning(
+                "Files from the build host were use to build this snap "
+                "but are not part of the snap or base: \n{}".format(formatted_system)
```

Comment:
I suggest "The {part_name!r} part needs the following libraries that are not included in the snap or base: \n{}\n\nThese dependencies can be satisfied via other stage-packages, other parts, or content sharing.".format(formatted_system)